applications/RomApplication/python_scripts/rom_manager.py

```diff
@@ -120,106 +99,99 @@ def Fit(self, mu_train=[None], mu_validation=[None], store_all_snapshots=False,
-                    if store_all_snapshots or store_rom_snapshots:
-                        self._StoreSnapshotsMatrix('rom_snapshots', rom_snapshots)
-                    self.ROMvsFOM_train = np.linalg.norm(fom_snapshots - rom_snapshots)/ np.linalg.norm(fom_snapshots)
+                    self._LaunchROM(mu_train)
+
                 if any(item == "HROM" for item in training_stages):
                     #FIXME there will be an error if we only train HROM, but not ROM
                     self._ChangeRomFlags(simulation_to_run = "trainHROMPetrovGalerkin")
-                    self.__LaunchTrainHROM(mu_train, store_residuals_projected)
+                    self._LaunchTrainHROM(mu_train)
                     self._ChangeRomFlags(simulation_to_run = "runHROMPetrovGalerkin")
-                    hrom_snapshots = self.__LaunchHROM(mu_train)
-                    if store_all_snapshots or store_hrom_snapshots:
-                        self._StoreSnapshotsMatrix('hrom_snapshots', hrom_snapshots)
-                    self.ROMvsHROM_train = np.linalg.norm(rom_snapshots - hrom_snapshots) / np.linalg.norm(rom_snapshots)
+                    self._LaunchHROM(mu_train)
             ##########################
         else:
             err_msg = f'Provided projection strategy {chosen_projection_strategy} is not supported. Available options are \'galerkin\', \'lspg\' and \'petrov_galerkin\'.'
             raise Exception(err_msg)
-
-    def StoreFomSnapshotsAndBasis(self, mu_train=[None]):
-        fom_snapshots = self.__LaunchTrainROM(mu_train)
-        self._StoreSnapshotsMatrix('fom_snapshots', fom_snapshots)
-
-    def StoreFomValidationSnapshots(self, mu_validation=[None]):
-        self.RunFOM(mu_run=mu_validation, snapshots_matrix_name='fom_snapshots_val')
-
-    def TrainAnnEnhancedROM(self):
-        self.__LaunchTrainNeuralNetwork()
+        if not type_of_decoder=="ann_enhanced":
+            #TODO implement online stage of ann_enhanced
+            self.ComputeErrors(mu_train)
+
+    def TrainAnnEnhancedROM(self, mu_train, mu_validation):
+        counter = 0
+        self.general_rom_manager_parameters["ROM"]["ann_enhanced_settings"]["training"]["model_number"].SetInt(counter)
+        in_database, _ = self.data_base.check_if_in_database("Neural_Network", mu_train)
+        if not in_database:
+            self._LaunchTrainNeuralNetwork(mu_train,mu_validation)
+        elif in_database and self.general_rom_manager_parameters["ROM"]["ann_enhanced_settings"]["training"]["retrain_if_exists"].GetBool():
+            while in_database:
+                counter+=1
+                self.general_rom_manager_parameters["ROM"]["ann_enhanced_settings"]["training"]["model_number"].SetInt(counter)
+                in_database, _ = self.data_base.check_if_in_database("Neural_Network", mu_train)
+            self.general_rom_manager_parameters["ROM"]["ann_enhanced_settings"]["training"]["model_number"].SetInt(counter)
```

Comment:
I think this line is not needed, as it's already done at the end of the loop

Reply:
true, erased